guides/dsl.md

```diff
@@ -0,0 +1,241 @@
+
+```javascript
+  program.block("block description", ({find, record}) => {
+    // search for records tagged "person"
+    let person = find("person");
+    return [
+      // bind a record tagged "greeting" with an attribute "message"
+      record("greeting", {message: `Hello ${person.name}`})
+    ];
+  });
+```
+
+The equivalent Eve block would be
+
+```eve
+search
+  person = [#person]
+
+bind
+  [#greeting message: "Hello {{person.name}}"]
+```
+
+The identity of a `record()` is established by everything within the parentheses. Therefore, the record `record("greeting", {message: `Hello ${person.name}`})` will create one record for every `person.name`.
+
+If would like to commit records instead of binding them, you can use the `commit()` method in place of `block()`. The implicatino here is that you can no longer create blocks which both commit and bind records at the same time, which we've found is not a common case.
+
+The DSL supports methods which mirror a couple of the Eve update operators. They are:
+
+### add()
+
+ dds attribute/value pairs to a record. These added values do not contribute to the identity of a record, allowing you to add multiple values toa single record. This is equivalent to the Eve `+=` operator. For example:
+
+```javascript
+program.block("every student is on the guest list", ({find, record}) => {
+  // search for records tagged "friend"
+  let friend = find("friend");
+  return [
+    // bind a single record tagged "guest-list", with every friend as a guest
+    record("guest-list").add({guest: friend})
+  ];
+});
+```
+
+### remove()
+
+Removes attribute/value pairs from a record. This is equivalent to the Eve `-=` oeprator. For example:
+
+```javascript
+program.block("blacklisted friends cannot come to the party", ({find, record}) => {
+  let blacklisted = find("blacklisted");
+  let guests = find("guest-list")
+  return [
+    // remove anyone blacklisted from the guestlist
+    guests.remove({guest: blacklisted})
+  ];
+});
+```
+
+You can also use `remove()` to completely delete records or attributes on a record. This is equivalent to `:= none` in Eve. For example:
+
+```javascript
+   // completely remove the target record
+   record.remove() 
+   // remove the attribute on the target record
+   record.remove("attribute")
+```
+
+The DSL does not contain an equivalent for the Eve general set operator `:=`, but you can mimick its behavior by chaining the `add()` and `remove()` operators. For example:
+
+```javascript
+program.block("Set Artemis' teacher", ({find, record}) => {
+  let artemis = find("student", {name: "Artemis"});
+  let teacher = find("teacher", {name: "Smith"});
+  return [
+    artemis.remove("teacher").add("teacher", teacher)
+  ];
+});
+```
+
+In summary:
+
+- Add a value: `person.salary += 10` <-> `person.add("salary", 10)`
```

Comment:
I think this section should just be a translation guide from the existing syntax to the JS DSL. We should add find and record here, along with the statements and expressions and aggregates.